unit test failure when run in isolation: views/sign_up->afterVisible->shows a tooltip...

I noticed this while trying to debug #6273 

> views/sign_up
>    afterVisible
>        shows a tooltip on the email element if model.bouncedEmail is set ‣
>        AssertionError: expected false to be true

blocks #6273

--- a/app/tests/lib/helpers.js
+++ b/app/tests/lib/helpers.js
@@ -66,11 +66,10 @@ define(function (require, exports, module) {
   function wrapAssertion(fn, done) {
     try {
       fn();
+      done();
     } catch (e) {
       done(e);
-      return;
     }
-    done();
   }
 
   function createRandomString(length, base = 36) {

--- a/app/tests/lib/helpers.js
+++ b/app/tests/lib/helpers.js
@@ -68,6 +68,7 @@ define(function (require, exports, module) {
       fn();
     } catch (e) {
       done(e);
+      return;
     }
     done();
   }

--- a/app/tests/spec/views/sign_up.js
+++ b/app/tests/spec/views/sign_up.js
@@ -287,10 +287,11 @@ define(function (require, exports, module) {
         view.render()
           .then(() => view.afterVisible())
           .then(function () {
-            assert.isTrue(view.showValidationError.called);
-            setTimeout(function () {
-              assert.isTrue(view.$('input[type="email"]').hasClass('invalid'));
-              done();
+            setTimeout(() => {
+              TestHelpers.wrapAssertion(function () {
+                assert.isTrue(view.showValidationError.called);
+                assert.isTrue(view.$('input[type="email"]').hasClass('invalid'));
+              }, done);
             }, 50);
           });
       });

